connection.js

```diff
@@ -153,20 +154,48 @@ function setupClient(self) {
     }
 }
 
-function Connection(client, server) {
+function Connection (client, server) {
     this.client = client;
     this.server = server;
-    this.local_ip = null;
-    this.local_port = null;
-    this.remote_ip = null;
-    this.remote_host = null;
-    this.remote_port = null;
-    this.remote_info = null;
+    this.local = {
+        ip: null,
+        proxy: null,
+        port: null,
+        host: null,
+    };
+    this.remote = {
+        ip:   null,
+        port: null,
+        host: null,
+        info: null,
+        closed: false,
+        is_private: false,
+    };
+    this.hello = {
+        host: null,
+        verb: null,
+    };
+    this.tls = {
+        enabled: server.has_tls ? true : false,
+        verified: false,
+        cipher: {},
+        authorized: null,
+    }
+
```

Comment:
Better for whom? Surely having a bunch of function calls is not going to be faster than just declaring the object and a few properties. I don't think that using .set() makes the code more readable for humans. In what way do you think it'd be better?